src/DependencyInjection/DI/src/ServiceLookup/CallSiteFactory.cs

```diff
@@ -415,6 +425,25 @@ public int Count
                 }
             }
 
+            public int GetSlot(ServiceDescriptor descriptor)
+            {
+                if (descriptor == _item)
+                {
+                    return 0;
+                }
+
+                if (_items != null)
+                {
+                    var index = _items.IndexOf(descriptor);
+                    if (index != -1)
+                    {
+                        return index + 1;
+                    }
+                }
+
+                throw new InvalidOperationException("Requested service descriptor doesn't exist.");
```

Comment:
Is this expected to be possible? Or should it be another Debug.Fail?

Reply:
Not expected but should fail in release too.